### ISSUE TITLE
IndexOutOfBoundsException for certain combination of template and data

### ISSUE THREAD

**Issue by mperktold** [spullara/mustache.java#291]
Consider the following code snippet:

```java
public static void main(String[] args) throws IOException {
    var factory = new DefaultMustacheFactory();
    try (var reader = new StringReader("{{No.}}");
        var writer = new StringWriter())
    {
        Mustache mustache = factory.compile(reader, "template");
        mustache.execute(writer, Map.of("No", "1"));
        System.out.println(writer);
    }
}
```

Notice there is a mismatch between template and data: the template uses a variable `No.`, i.e. with a trailing dot, whereas the data provides a variable `No` without a dot.

When executing this code, an Exception is thrown:

```
Exception in thread "main" com.github.mustachejava.MustacheException: Failed to get value for No. @[template:1]
	at com.github.mustachejava.codes.ValueCode.execute(ValueCode.java:74)
	at com.github.mustachejava.codes.DefaultMustache.run(DefaultMustache.java:34)
	at com.github.mustachejava.codes.DefaultCode.execute(DefaultCode.java:162)
	at com.github.mustachejava.codes.DefaultMustache.execute(DefaultMustache.java:57)
	at com.github.mustachejava.Mustache.execute(Mustache.java:38)
	at it.prodata.gest.datahub.MustacheTemplateEngine.main(MustacheTemplateEngine.java:64)
Caused by: com.github.mustachejava.MustacheException: begin 0, end 1, length 0 @[template:1]
	at com.github.mustachejava.codes.DefaultCode.get(DefaultCode.java:150)
	at com.github.mustachejava.codes.ValueCode.execute(ValueCode.java:62)
	... 5 more
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 1, length 0
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4604)
	at java.base/java.lang.String.substring(String.java:2707)
	at com.github.mustachejava.reflect.BaseObjectHandler.findMemberOnClass(BaseObjectHandler.java:162)
	at com.github.mustachejava.reflect.BaseObjectHandler.findMember(BaseObjectHandler.java:154)
	at com.github.mustachejava.reflect.ReflectionObjectHandler.findWrapper(ReflectionObjectHandler.java:156)
	at com.github.mustachejava.reflect.ReflectionObjectHandler.find(ReflectionObjectHandler.java:114)
	at com.github.mustachejava.reflect.GuardedBinding.getWrapper(GuardedBinding.java:103)
	at com.github.mustachejava.reflect.GuardedBinding.createAndGet(GuardedBinding.java:89)
	at com.github.mustachejava.reflect.GuardedBinding.get(GuardedBinding.java:84)
	at com.github.mustachejava.codes.DefaultCode.get(DefaultCode.java:145)
	... 6 more
```

The important part is `BaseObjectHandler.findMemberOnClass`. At some point, it tries to find a getter for the variable name, and thus changes the first character to uppercase. And that fails with an `IndexOutOfBoundsException` for an empty variable name.

This shouldn't happen, since the method clearly tries to use several strategies and to return null if nothing works. Failing with an exception seems not intended. Also, the same template does work when I pass `Map.of("No", Map.of("", 1))` as data.

I stumbled upon this when changing a variable name from something like `No.` to `No`, i.e. removing the trailing dot. I had some old templates around, and when I rendering the old template with the new data, this Exception was thrown.

**Comment by spullara** [comment]
This does look like a case where I would have expected it just to return null. Thanks for the report.

**Comment by spullara** [comment]
Fixed 304f71a02f7a6be5dec373e0cd00eab3de11d8f3
